lib/lighthouse/benefits_documents/service.rb

```diff
@@ -0,0 +1,110 @@
+# frozen_string_literal: true
+
+require 'common/client/base'
+require 'lighthouse/benefits_documents/configuration'
+require 'lighthouse/benefits_documents/service_exception'
+require 'lighthouse/service_exception'
+
+module BenefitsDocuments
+  class Service < Common::Client::Base
+    configuration BenefitsDocuments::Configuration
+    STATSD_KEY_PREFIX = 'api.benefits_documents'
+    STATSD_UPLOAD_LATENCY = 'lighthouse.api.benefits.documents.latency'
+
+    def initialize(icn)
+      @icn = icn
+      raise ArgumentError, 'no ICN passed in for LH API request.' if icn.blank?
+
+      super()
+    end
+
+    def queue_document_upload(params, lighthouse_client_id = nil)
+      start_timer = Time.zone.now
+      claim_id = params[:claimId]
+      raise Common::Exceptions::RecordNotFound, claim_id unless claim_id
```

Comment:
To me this seems more like an `ArgumentError` vs a `RecordNotFound` error since it looks like there's no record lookup here. The way the service currently works is that the controller is the one responsible for raising a record not found error if it can't find an EVSS claim with the given claim ID.

Reply:
That's a good call. I was torn because it is tied to a record, but you're right that it's not a failed lookup that would cause this error. Updated